panels/class-debug-bar-deprecated.php

```diff
@@ -209,4 +227,49 @@ public static function deprecated_argument_run( $function, $message, $version )
 
 		error_log( 'Deprecation Notice: ' . strip_tags( $message ) . '  in ' . $location );
 	}
+
+	/**
+	 * Log notices about deprecated (PHP 4) class constructors being run.
+	 *
+	 * @since 0.10.0
+	 *
+	 * @param string $class        The class containing the deprecated constructor.
+	 * @param string $version      The version of WordPress that deprecated the function.
+	 * @param string $parent_class Optional. The parent class calling the deprecated constructor.
+	 */
+	public static function deprecated_constructor_run( $class, $version, $parent_class = '' ) {
+		$backtrace = debug_backtrace( false );
+		$bt        = 4;
+		if ( ! isset( $backtrace[4]['file'] ) && 'call_user_func_array' === $backtrace[5]['function'] ) {
+			$bt = 6;
+		}
+
+		$file     = ( isset( $backtrace[ $bt ]['file'] ) ? $backtrace[ $bt ]['file'] : 0 );
+		$line     = ( isset( $backtrace[ $bt ]['line'] ) ? $backtrace[ $bt ]['line'] : 0 );
+		$location = $file . ':' . $line;
+
+		if ( ! empty( $parent_class ) ) {
+			$message = sprintf(
+				/* translators: 1: PHP class name, 2: PHP parent class name, 3: version number, 4: __construct() method */
+				__( 'The called constructor method for %1$s in %2$s is <strong>deprecated</strong> since version %3$s! Use %4$s instead.', 'debug-bar' ),
+				$class,
+				$parent_class,
+				$version,
+				'<pre>__construct()</pre>'
```

Comment:
Can we use `code` tags here instead since they're inline?

Reply:
Indeed, that would be the better tag. Fixed.

Reply:
As a side-note: the basis of the code was copied from WP core, so this may be a change you should also suggest to be made in core:
* https://developer.wordpress.org/reference/functions/_deprecated_constructor/
* https://core.trac.wordpress.org/browser/trunk/src/wp-includes/functions.php#L3902

Reply:
Nice catch! Sure, let's get that done at contributor day at WordCamp Nijmegen this weekend